### ISSUE TITLE
Error thrown when XHR requests are aborted

### ISSUE THREAD

**Issue by begizi** [cypress-io/cypress#62]
The application being tested will abort a request when multiple components change state before the previous xhr request is finished loading and is throwing this error on the request that gets aborted.

```
Error: INVALID_STATE_ERR - 0
    at verifyRequestOpened (http://localhost:2020/__cypress/static/js/sinon.js:4640:19)
    at FakeXMLHttpRequest.setResponseHeaders (http://localhost:2020/__cypress/static/js/sinon.js:4796:17)
    at FakeXMLHttpRequest.respond (http://localhost:2020/__cypress/static/js/sinon.js:4928:22)
    at http://localhost:2020/__cypress/static/js/cypress.js:3346:27
    at i (http://localhost:2020/__cypress/static/js/vendor.js:565:40)
    at e.exports.e._settlePromiseFromHandler (http://localhost:2020/__cypress/static/js/vendor.js:564:5292)
    at e.exports.e._settlePromiseAt (http://localhost:2020/__cypress/static/js/vendor.js:564:6561)
    at e.exports.e._settlePromises (http://localhost:2020/__cypress/static/js/vendor.js:564:8426)
    at i._drainQueue (http://localhost:2020/__cypress/static/js/vendor.js:562:3183)
    at i._drainQueues (http://localhost:2020/__cypress/static/js/vendor.js:562:3244)
    at MutationObserver.drainQueues (http://localhost:2020/__cypress/static/js/vendor.js:562:1186)
```

**Comment by brian-mann** [comment]
Yup this is a known issue: https://github.com/cypress-io/cypress/issues/34

When we talked earlier this is what I was describing when I talked about refactoring XHR's. These are bugs in the internals of sinon and all of this code needs to be removed.

XHR mocking will move into the server implementation behind the scenes and there will only need to be a very lightweight object for handling XHR + window.fetch. This will alleviate the need to implement the full XHR + fetch API (which sinon attempts to do).

By allowing the actual requests to leave the browser there should be less edge cases, less chance of not implementing the objects with 100% fidelity, and it will make Cypress more future proof.
